### ISSUE TITLE
Run-to-run variability

### ISSUE THREAD

**Issue by ddpinto** [davidaknowles/leafcutter#33]
Hello,
Running the leafcutter differential splicing analysis multiple times on the same case/control dataset using the exact same parameters and cluster count inputs, gives varying loglr and p value results between runs. This happens irrespective of using covariates or not. 

Could you perhaps explain why the results differ per run? Is there a permutation or estimation step that causes the variation and, if so, can it be fixed by setting a particular seed value?

Thanks!

**Comment by davidaknowles** [comment]
There's no permutation but the optimizer is given a random initialization, which may not be ideal. I'll look into providing a sensible deterministic intialization. Out of curiosity how inconsistent are we talking? I would hope it wouldn't be too bad but it's not a convex problem so there's no guarantee of getting to the same optimum on a second run.

**Comment by davidaknowles** [comment]
I've looked our 4 YRI vs 4 EU test and there is some variability (this is p-values for two runs)
![image](https://user-images.githubusercontent.com/4521734/30083256-a3d89ecc-9242-11e7-981e-413779e33d0b.png)

It looks like the majority of clusters get to the same solution but for some a different solution is found. Simply setting the initialization to a zero vector performs badly so I'll try to come up with something better. Failing that I'll add an option to set the seed.

**Comment by davidaknowles** [comment]
Ok so I've added "smart" deterministic initialization. This is a one-shot method of moments estimator for the null model regression coefficients, and I just set the initial concentration parameter(s) to 10 which should be reasonable. The agreement between this and a random init seems similar to that between two random inits:
![image](https://user-images.githubusercontent.com/4521734/30086568-8ecc335a-9250-11e7-88d6-5edaaf5ac2cd.png)
I've tested this on the 4 YRI vs 4 EUR chr1 example but in case there are unforeseen issues on other datasets I've added an option (--init) to revert back to random initialization and set the random seed.
